pkg/catalog/loader/loader.go

```diff
@@ -528,6 +549,55 @@ func (store *Store) LoadTemplatesWithTags(templatesList, tags []string) []*templ
 	return loadedTemplates.Slice
 }
 
+var (
+	numericalExpressionRegex = regexp.MustCompile(`^[0-9+\-/\W]+$`)
+	unresolvedVariablesRegex = regexp.MustCompile(`(?:%7[B|b]|\{){2}([^}]+)(?:%7[D|d]|\}){2}["'\)\}]*`)
+)
+
+// copy of the original function from pkg/protocols/common/expressions/variables.go:ContainsUnresolvedVariables
+func templateContainsUnresolvedVariables(templatePath string) bool {
+	data, err := os.ReadFile(templatePath)
+	if err != nil {
+		return false
+	}
+
+	matches := unresolvedVariablesRegex.FindAllStringSubmatch(string(data), -1)
+	if len(matches) == 0 {
+		return false
+	}
+
+	var unresolvedVariables []string
+	for _, match := range matches {
+		if len(match) < 2 {
+			continue
+		}
+
+		// Skip if the match is an expression
+		if numericalExpressionRegex.MatchString(match[1]) {
+			continue
+		}
+		// or if it contains only literals (can be solved from expression engine)
+		if hasLiteralsOnly(match[1]) {
+			continue
+		}
+		unresolvedVariables = append(unresolvedVariables, match[1])
+	}
+
+	return len(unresolvedVariables) > 0
+}
+
+func hasLiteralsOnly(data string) bool {
+	expr, err := govaluate.NewEvaluableExpressionWithFunctions(data, dsl.HelperFunctions)
+	if err != nil {
+		return false
+	}
+	if expr != nil {
+		_, err = expr.Evaluate(nil)
+		return err == nil
+	}
```

Comment:
if this is duplicated i think we can move it to pkg/utils because to avoid any inconsistencies in future and we can reuse it for other purpose as well